### DISCUSSION TITLE
Issue with bump-cask-pr wiso-steuer-2024

### DISCUSSION THREAD

**Discussion by joh-klein** [Homebrew/discussions#5548] [Everyday usage] [answered]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.3.13-31-g90a90b3
ORIGIN: https://github.com/Homebrew/brew
HEAD: 90a90b3076910375fa05e99197950cf2846bc853
Last commit: 7 hours ago
Core tap HEAD: 9a00a903624a87c016d99f9d09a3c1b419573aa0
Core tap last commit: 77 minutes ago
Core tap JSON: 07 Aug 13:41 UTC
Core cask tap HEAD: 0aef0ff520b1ed766d94ce8cdd0d964ac35e38d3
Core cask tap last commit: 2 hours ago
Core cask tap JSON: 07 Aug 13:41 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_GITHUB_API_TOKEN: set
HOMEBREW_MAKE_JOBS: 10
HOMEBREW_SORBET_RUNTIME: set
Homebrew Ruby: 3.3.4 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.3.4_1/bin/ruby
CPU: 10-core 64-bit arm_firestorm_icestorm
Clang: 15.0.0 build 1500
Git: 2.39.3 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 8.1.2 => /usr/bin/curl
macOS: 13.6.3-arm64
CLT: 15.1.0.0.1.1700200546
Xcode: N/A
Rosetta 2: false
```

### Output of `brew doctor`

```shell
not applicable
```

### Description of issue

I wanted to update the `wiso-steuer-2024` cask but there are issues:

`brew bump-cask-pr wiso-steuer-2024 --version 31.09.4100`

The version in the appcast.xml looks like this:

```
<enclosure
url="https://update.buhl-data.com/Updates/Steuer/2024/Mac/Files/31.09.4100-RC1/SteuerMac2024-31.09.4100.dmg"
sparkle:version="4100"
sparkle:shortVersionString="31.09.4100"
sparkle:edSignature="j6nZMSPCRf519f3sxM4a//xNg5QY16Sqnge3sXiJ1VLBHSC0IzvY6+EwVwukwpsWmSWCZlItFsbj6/GaQOURCQ=="
length="753598567"
type="application/octet-stream"/>
```

But [the cask](https://github.com/Homebrew/homebrew-cask/blob/HEAD/Casks/w/wiso-steuer-2024.rb) only wants to use

`  url "https://update.buhl-data.com/Updates/Steuer/2024/Mac/Files/#{version}-HF1/SteuerMac2024-#{version}.dmg",`

tl;dr: They are changing "a little more" in the url then the version. How can I update the casks that future bump-casks work without an issue?

**Comment by p-linnane** [comment, marked as answer]
We can capture the `-HF1` or `-RC1` as part of the version, and insert that into that first part of the URL. I'll open a PR to fix this.

**Reply by p-linnane** [reply]
https://github.com/Homebrew/homebrew-cask/pull/181821
